### ISSUE TITLE
Create Order via REST API adds tax to shipping

### ISSUE THREAD

**Issue by singhdd93** [woocommerce/woocommerce#13052]
## EXPLANATION OF THE ISSUE
Creating an Order via REST API adds tax to shipping as well. Here is json posted for creating order
```
{
  "payment_method": "paypal_express",
  "payment_method_title": "Paypal Express",
  "set_paid": false,
  "customer_id": 2,
  "billing": {
    "first_name": "Damandeep",
    "last_name": "Singh",
    "address_1": "1257 ",
    "address_2": "Coach House Court",
    "city": "Fullerton",
    "state": "CA",
    "postcode": "92831",
    "country": "US",
    "email": "daman@singhdd.com",
    "phone": "(122) 274-5555"
  },
  "shipping": {
    "first_name": "Damandeep",
    "last_name": "Singh",
    "address_1": "1257",
    "address_2": "Coach House Court",
    "city": "Fullerton",
    "state": "CA",
    "postcode": "92831",
    "country": "US"
  },
  "line_items": [
    {
      "product_id": 531,
      "variation_id": 533,
      "quantity": 1
    }
  ],
  "shipping_lines": [
    {
      "method_id": "usps_medium_box",
      "method_title": "USPS Medium Flat Rate Box",
      "total": 13.60
    }
  ]
}
```

Here is list of taxes
```
[
  {
    "id": 1,
    "country": "",
    "state": "",
    "postcode": "",
    "city": "",
    "rate": "9.0000",
    "name": "Sales Tax",
    "priority": 1,
    "compound": false,
    "shipping": false,
    "order": 0,
    "class": "standard",
    "_links": {
      "self": [
        {
          "href": "https://ambala.webdemos.cf/wp-json/wc/v1/taxes/1"
        }
      ],
      "collection": [
        {
          "href": "https://ambala.webdemos.cf/wp-json/wc/v1/taxes"
        }
      ]
    }
  }
]
```

Here is response of creating order
```
{
  "id": 605,
  "parent_id": 0,
  "status": "pending",
  "order_key": "wc_order_5893614a8fb78",
  "number": 605,
  "currency": "USD",
  "version": "2.6.13",
  "prices_include_tax": false,
  "date_created": "2017-02-02T16:41:46",
  "date_modified": "2017-02-02T16:41:46",
  "customer_id": 2,
  "discount_total": "0.00",
  "discount_tax": "0.00",
  "shipping_total": "13.60",
  "shipping_tax": "1.22",
  "cart_tax": "1.44",
  "total": "32.24",
  "total_tax": "2.66",
  "billing": {
    "first_name": "Damandeep",
    "last_name": "Singh",
    "company": "",
    "address_1": "1257 ",
    "address_2": "Coach House Court",
    "city": "Fullerton",
    "state": "CA",
    "postcode": "92831",
    "country": "US",
    "email": "daman@singhdd.com",
    "phone": "(122) 274-5555"
  },
  "shipping": {
    "first_name": "Damandeep",
    "last_name": "Singh",
    "company": "",
    "address_1": "1257",
    "address_2": "Coach House Court",
    "city": "Fullerton",
    "state": "CA",
    "postcode": "92831",
    "country": "US"
  },
  "payment_method": "paypal_express",
  "payment_method_title": "Paypal Express",
  "transaction_id": "",
  "customer_ip_address": "103.41.36.35",
  "customer_user_agent": "PostmanRuntime/3.0.9",
  "created_via": "rest-api",
  "customer_note": "",
  "date_completed": "2017-02-02T08:41:46",
  "date_paid": "",
  "cart_hash": "",
  "line_items": [
    {
      "id": 79,
      "name": "Kaju Katli",
      "sku": "SW-282",
      "product_id": 491,
      "variation_id": 494,
      "quantity": 1,
      "tax_class": "",
      "price": "15.98",
      "subtotal": "15.98",
      "subtotal_tax": "1.44",
      "total": "15.98",
      "total_tax": "1.44",
      "taxes": [
        {
          "id": 1,
          "total": 1.4382,
          "subtotal": 1.4382
        }
      ],
      "meta": [
        {
          "key": "packing-size",
          "label": "Packing Size",
          "value": "2lb Box"
        }
      ]
    }
  ],
  "tax_lines": [
    {
      "id": 81,
      "rate_code": "SALES TAX-1",
      "rate_id": "1",
      "label": "Sales Tax",
      "compound": false,
      "tax_total": "1.44",
      "shipping_tax_total": "1.22"
    }
  ],
  "shipping_lines": [
    {
      "id": 80,
      "method_title": "USPS Medium Flat Rate Box",
      "method_id": "usps_medium_box",
      "total": "13.60",
      "total_tax": "0.00",
      "taxes": []
    }
  ],
  "fee_lines": [],
  "coupon_lines": [],
  "refunds": [],
  "_links": {
    "self": [
      {
        "href": "https://ambala.webdemos.cf/wp-json/wc/v1/orders/605"
      }
    ],
    "collection": [
      {
        "href": "https://ambala.webdemos.cf/wp-json/wc/v1/orders"
      }
    ],
    "customer": [
      {
        "href": "https://ambala.webdemos.cf/wp-json/wc/v1/customers/2"
      }
    ]
  }
}
```

## STEPS TO REPRODUCE THE ISSUE

Create a order via REST API

## SYSTEM STATUS REPORT

```
`
### WordPress Environment ###

Home URL: https://ambala.webdemos.cf
Site URL: https://ambala.webdemos.cf
WC Version: 2.6.13
Log Directory Writable: ✔
WP Version: 4.7.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache/2.4.7
PHP Version: 5.6.30-1+deb.sury.org~trusty+1
PHP Post Max Size: 8 MB
PHP Time Limit: 599
PHP Max Input Vars: 1000
cURL Version: 7.35.0
OpenSSL/1.0.1f

SUHOSIN Installed: –
MySQL Version: 5.6.33
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 2.6.13
: 
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Active Plugins (29) ###

Redux Framework: by Team Redux – 3.6.3
JSON Basic Authentication: by WordPress API Team – 0.1
AJAX Heartbeat Tool: by Mikel King – 1.4
Archives Calendar Widget: by Aleksei Polechin (alek´) – 1.0.12
CMB2: by WebDevStudios – 2.2.3.1
Contact Form 7: by Takayuki Miyoshi – 4.6.1
Easy Theme and Plugin Upgrades: by Chris Jean – 2.0.0
Envato WordPress Toolkit: by Envato – 1.7.3
Events Manager: by Marcus Sykes – 5.6.6.1
Find and Replace All: by Taraprasad Swain – 1.1
Import Majesty XML Demo: by samathemes – 1.0.0
WPBakery Visual Composer: by Michael M - WPBakery.com – 5.0.1
Custom Login Page Customizer: by Hardeep Asrani – 1.1.0
Majesty Shortcodes: by samathemes – 1.0
Max Mega Menu: by Tom Hemsley – 2.3.4
Our Team: by WooThemes – 1.4.1
PayPal for WooCommerce: by Angell EYE – 1.3.2
WP REST API: by WP REST API Team – 2.0-beta15
Restaurant Reservations: by Theme of the Crop – 1.7.2
WooCommerce PDF Invoices & Packing Slips: by Ewout Fernhout – 1.5.38
WooCommerce Simple Registration: by Astoundify – 1.3.0
WooCommerce US Postal Service Webservice Method: by Jamez Picard – 1.2.7
WooCommerce: by WooThemes – 2.6.13
WooSidebars: by WooThemes – 1.4.3
WP REST Api Extras: by Jan-Louis Crafford
Afrozaar Consulting – 0.23

WP Clone by WP Academy: by WP Academy – 2.2.4
WP FullCalendar: by Marcus Sykes – 1.2
WP Smush: by WPMU DEV – 2.5.3
WP Speed Grades Lite: by Bestseogr – 2.0

### Settings ###

Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2

### API ###

API Enabled: ✔

### WC Pages ###

Shop Base: #81 - /shop/
Cart: #82 - /cart/
Checkout: #83 - /checkout/
My Account: #84 - /my-account/

### Taxonomies ###

Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)


### Theme ###

Name: Majesty Child
Version: 1.0
Author URL: http://samathemes.com
Child Theme: ✔
Parent Theme Name: Majesty
Parent Theme Version: 1.4.1
Parent Theme Author URL: http://themeforest.net/user/samathemes
WooCommerce Support: ✔

### Templates ###

Overrides: majesty/woocommerce/global/breadcrumb.php
majesty/woocommerce/global/quantity-input.php
majesty/woocommerce/loop/add-to-cart.php
majesty/woocommerce/loop/loop-end.php
majesty/woocommerce/loop/loop-start.php
majesty/woocommerce/product-searchform.php
majesty/woocommerce/single-product/review-meta.php

`
```
##Note
Creating order via website is fine
Order created via REST API can be fixed by pressing "Calculate Tax" & "Calculate Total" buttons on order details page in wp admin

**Comment by mikejolley** [comment]
This will be fixed in 2.7 - already confirmed it's not an issue in that version because the calculation logic has been merged with that in admin.

**Comment by nilaybrahmbhatt** [comment]
how to pass taxes slabs in params to add taxes in create order api?
